transport/transport_test.go

```diff
@@ -732,7 +732,7 @@ func TestEncodingRequiredStatus(t *testing.T) {
 		Last:  true,
 		Delay: false,
 	}
-	if err := ct.Write(s, expectedRequest, &opts); err != nil {
+	if err := ct.Write(s, expectedRequest, &opts); err != nil || err == io.EOF {
```

Comment:
this change is nonsense.

if `err == io.EOF` then `err != nil` is already true. Perhaps you meant `err != nil && err != io.EOF`?